crates/swc_ecma_minifier/src/compress/optimize/unused.rs

```diff
@@ -313,7 +313,7 @@ where
 
             Expr::Paren(p) => return self.should_preserve_property_access(&p.expr, opts),
 
-            Expr::Fn(..) | Expr::Arrow(..) | Expr::Array(..) => {
+            Expr::Fn(..) | Expr::Arrow(..) | Expr::Array(..) | Expr::Class(..) => {
```

Comment:
Accessing a property of a class can have a side-effect.

Reply:
Nevermind I found the logic below

Reply:
This is actually wrong

Reply:
Is this code fine?

Reply:
I reviewed every other thing including tsc tests, but still worried about this.

Reply:
Yes, I have patched may_have_side_effect, it's fixed now.